src/main/java/seedu/address/logic/commands/WhyCommand.java

```diff
@@ -35,19 +41,19 @@ public WhyCommand(Index targetIndex) {
     }
 
     @Override
-    public CommandResult execute() {
-        //EventsCenter.getInstance().post(new ShowHelpRequestEvent());
+    public CommandResult execute() throws CommandException {
 
         List<ReadOnlyPerson> lastShownList = model.getFilteredPersonList();
 
         if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            //throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
 
-        ReadOnlyPerson personToDelete = lastShownList.get(targetIndex.getZeroBased());
-        Name name = personToDelete.getName();
-        Address address = personToDelete.getAddress();
-        String reason = personToDelete.getReason();
+
+        ReadOnlyPerson personToAnswer = lastShownList.get(targetIndex.getZeroBased());
+        Name name = personToAnswer.getName();
+        Address address = personToAnswer.getAddress();
+        String reason = personToAnswer.getReason();
         //return new CommandResult(String.format(SHOWING_WHY_MESSAGE, name, address));
```

Comment:
please also clean your code of comments, me and oscar had to help you clean up several of these. You should not be leaving grey code around if they dont give insight to people who are taking over your code